packages/rich-text/src/component/use-anchor.js

```diff
@@ -174,7 +172,8 @@ export function useAnchor( { editableContentElement, settings = {} } ) {
 			editableContentElement.removeEventListener( 'focusin', attach );
 			editableContentElement.removeEventListener( 'focusout', detach );
 		};
-	}, [ editableContentElement, tagName, className ] );
+	}, [ editableContentElement, tagName, className, callback ] );
 
+	anchor.update = callback;
 	return anchor;
```

Comment:
We probably can't really get away with modifying the API in this way. I expect we will have to create a complementary hook which basically does the same thing as `useAnchor` but has a different "export" signature. 

For example:

```
const { anchor, updateAnchor } = useMyNewAnchor( args );
```

Thoughts welcomed.

Reply:
I suppose we can't modify the return of `useAnchor`?

Reply:
It's a public API so no.

I think we'll need a new hook.

We could pass additional dependencies into the hook to trigger the `useLayoutEffect` to recompute the value but that would couple the implementation to specific dependencies.

I think exposing the update function is the simplest IoC technique.

Reply:
I'm going to create a new base hook called `useAnchorBase` which will basically be `useAnchor` but it will export an object with the anchor and update method.

Then I'll rewrite `useAnchor` to call that hook but only return the `anchor` (as it does currently).

Then I"ll create a new private hook to also use the `useAnchorBase` but it will return the entire object.

Then I'll consume the private hook in this PR.

Reply:
I've now exposed a private API. Let me know what you think @scruffian.

Reply:
Actually I found an even better way which is standardised

https://github.com/WordPress/gutenberg/pull/57726/commits/160f4d3c46bae1f24b89dc4473e754b36c5c6622